templates/aws-vpc.template.yaml

```diff
@@ -340,6 +395,29 @@ Conditions:
   PrivateSubnets&4AZCondition: !And
     - !Condition 'PrivateSubnetsCondition'
     - !Condition '4AZCondition'
+  PrivateSubnets&PublicSubnets&NatGatewaysCondition: !And
+    - !Condition 'PrivateSubnetsCondition'
+    - !Condition 'PublicSubnetsCondition'
+    - !Condition 'NATGatewaysCondition'
+  PrivateSubnets&PublicSubnets&NatGateways&3AZCondition: !And
```

Comment:
On the similar note as abvoe comment, this condition is redundant of Nategtway&3AZCondition.